src/dftbp/dftb/periodic.F90

```diff
@@ -898,11 +898,11 @@ subroutine fillNeighbourArrays(neigh, iNeighbour, neighDist2, startAtom, endAtom
       call neigh%neighDist2Win_%unlock()
     #:endif
     else
-      if (associated(neigh%iNeighbourMem_)) then
-        deallocate(neigh%iNeighbourMem_)
+      if (associated(neigh%iNeighbour)) then
+        deallocate(neigh%iNeighbour)
       end if
-      if (associated(neigh%neighDist2Mem_)) then
-        deallocate(neigh%neighDist2Mem_)
+      if (associated(neigh%neighDist2)) then
+        deallocate(neigh%neighDist2)
       end if
 
       allocate(neigh%iNeighbour(0:maxNeighbour, 1:nAtom))
```

Comment:
I would argue that we could delete the entire deallocation-allocation from this routine, because there is `allocateNeighbourArrays` being called right before it. Could you please check that?

Reply:
Now removed.